garage/spaces/box.py

```diff
@@ -33,16 +33,11 @@ def __init__(self, low, high, shape=None, dtype=None):
             self.low = low + np.zeros(shape)
             self.high = high + np.zeros(shape)
 
-        if dtype is None:
-            if (self.low == 0).all() and (self.high == 255).all():
-                self.dtype = np.uint8
-                warnings.warn("garage.spaces.Box detected dtype as np.uint8. "
-                              "Please provide explicit dtype.")
-            else:
-                self.dtype = np.float32
+        if (self.low == 0).all() and (self.high == 255).all():
+            warnings.warn("garage.spaces.Box detected dtype as np.uint8. "
```

Comment:
This should probably just read: `creating a garage.spaces.Box with low = 0, high = 255, and dtype = np.float32`, since nothing's really detected and following the instructions won't silence the warning.

Also, the warning shouldn't be triggered if `dtype == np.uint8`.